capstone-frontend/src/components/ListPage.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+
+class ListPage extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      gBooks: [],
+      loading: true,
+      empty: false
+    }
+  }
+
+
+  fetchBooks = async () => {
+
+    const userID = window.localStorage.getItem("userID");
+
+    console.log("Fetch Books LIST PAGE")
+
+    const bookLists = await fetch(`/api/booklist?userID=${userID}`, {
+      method: "GET",
+    }).then(resp => resp.json());
+
+    console.log("Booklist", bookLists)
+
+    const bookList = Array.from(bookLists).filter(bookList => (bookList.id === this.props.match.params.id))[0].gbookIDs;
```

Comment:
+1. This may also cut down on the time needed to send and receive the request, and you can potentially eliminate your "loading page" that you implemented due to long fetch times. It would definitely be quicker to just fetch a single BookList rather than fetch all of a user's lists, and then filter for the one you want. I know much of the latency comes from fetching each individual book on the list, but this could help as well.

Reply:
Also, if you use the generalized backend functions, this behavior is already built. So it should be an easy switch.

Reply:
Done.